api/config/v1alpha1/envoygateway_types.go

```diff
@@ -177,15 +177,37 @@ type EnvoyGatewayKubernetesProvider struct {
 	OverwriteControlPlaneCerts bool `json:"overwrite_control_plane_certs,omitempty"`
 }
 
+const (
+	// KubernetesWatchModeTypeNamespaces indicates that the namespace watch mode is used.
+	KubernetesWatchModeTypeNamespaces = "Namespaces"
+
+	// KubernetesWatchModeTypeNamespaceSelectors indicates that namespaceSelectors watch
+	// mode is used.
+	KubernetesWatchModeTypeNamespaceSelectors = "NamespaceSelectors"
+)
+
+// KubernetesWatchModeType defines the type of KubernetesWatchMode
+type KubernetesWatchModeType string
+
 // KubernetesWatchMode holds the configuration for which input resources to watch and reconcile.
 type KubernetesWatchMode struct {
+	// Type indicates what watch mode to use. KubernetesWatchModeTypeNamespaces and
+	// KubernetesWatchModeTypeNamespaceSelectors are currently supported
+	// By default, when this field is unset or empty, Envoy Gateway will watch for input namespaced resources
+	// from all namespaces.
```

Comment:
needs a default tag here similar to https://github.com/envoyproxy/gateway/blob/91c0888ad63631440286264f18f1bd8d2ccab4cd/api/config/v1alpha1/envoyproxy_types.go#L169

Reply:
Do I miss anything here? This is an optional field. It has to be defined with either `Namespaces` or `NamespaceSelectors`

Reply:
Hi @arkodg, do you mind helping me understand this?

Reply:
this should be `// +kubebuilder:default=Namespaces`